bin/spatially_variable_genes.qmd

```diff
@@ -0,0 +1,141 @@
+---
+title: "nf-core/spatialtranscriptomics"
+subtitle: "Neighborhood enrichment analysis and Spatially variable genes"
+format:
+    nf-core-html: default
+jupyter: python3
+---
+
+```{python}
+#| tags: [parameters]
+#| echo: false
+input_sdata = "sdata_processed.zarr"  # Input: SpatialData file
+n_top_spatial_degs = 14  # Number of spatially variable genes to plot
+artifact_dir = "artifacts"  # Output directory
+output_csv = "spatially_variable_genes.csv"  # Output: gene list
+output_adata = "adata_spatially_variable_genes.h5ad"  # Output: AnnData file
+output_sdata = "sdata.zarr"  # Output: SpatialData file
+```
+
+```{python}
+import numpy as np
+import os
+import pandas as pd
+import scanpy as sc
+import squidpy as sq
+import spatialdata
+from anndata import AnnData
+from matplotlib import pyplot as plt
+```
+
+```{python}
+# Make sure we can use scanpy plots with the AnnData object exported from sdata.table
+# This code is taken from the early version of https://github.com/scverse/spatialdata-io/pull/102/
+# Once the PR will be merged in spatialdata-io, we should use spatialdata_io.to_legacy_anndata(sdata).
+def to_legacy_anndata(sdata: spatialdata.SpatialData) -> AnnData:
+    adata = sdata.table
+    for dataset_id in adata.uns["spatial"]:
+        adata.uns["spatial"][dataset_id]["images"] = {
+            "hires": np.array(sdata.images[f"{dataset_id}_hires_image"]).transpose([1, 2, 0]),
+            "lowres": np.array(sdata.images[f"{dataset_id}_lowres_image"]).transpose([1, 2, 0]),
+        }
+        adata.uns["spatial"][dataset_id]["scalefactors"] = {
+            "tissue_hires_scalef": spatialdata.transformations.get_transformation(
+                sdata.shapes[dataset_id], to_coordinate_system="downscaled_hires"
+            ).scale[0],
+            "tissue_lowres_scalef": spatialdata.transformations.get_transformation(
+                sdata.shapes[dataset_id], to_coordinate_system="downscaled_lowres"
+            ).scale[0],
+            "spot_diameter_fullres": sdata.shapes[dataset_id]["radius"][0] * 2,
+        }
+    return adata
+```
+
+```{python}
+# Read data
+sdata = spatialdata.read_zarr("./" + input_sdata, ["images", "table", "shapes"])
+
+adata = to_legacy_anndata(sdata)
+print("Content of the AnnData object:")
+print(adata)
+
+# Fix for scanpy issue https://github.com/scverse/scanpy/issues/2181
```

Comment:
Is this still required? Apparently this was fixed; https://github.com/scverse/scanpy/issues/2181#issuecomment-1640066417.

Reply:
You are right, it was fixed in the last release, so I removed the fix in https://github.com/nf-core/spatialtranscriptomics/commit/a9a55705eca90389ff3137b522c050746c5563f5.